src/sql/workbench/common/actions.ts

```diff
@@ -239,16 +239,23 @@ export class BackupAction extends Task {
 	}
 
 	runTask(accessor: ServicesAccessor, profile: IConnectionProfile): void | Promise<void> {
+		let configurationService = accessor.get<IConfigurationService>(IConfigurationService);
+		let previewFeaturesEnabled: boolean = configurationService.getValue('workbench')['enablePreviewFeatures'];
+		if (!previewFeaturesEnabled) {
+			return accessor.get<INotificationService>(INotificationService).info(nls.localize('backup.isPreviewFeature', 'You must enable preview features in order to use backup'));
+		}
+
+		let connectionManagementService = accessor.get<IConnectionManagementService>(IConnectionManagementService);
 		if (!profile) {
 			let objectExplorerService = accessor.get<IObjectExplorerService>(IObjectExplorerService);
-			let connectionManagementService = accessor.get<IConnectionManagementService>(IConnectionManagementService);
 			let workbenchEditorService = accessor.get<IEditorService>(IEditorService);
 			profile = TaskUtilities.getCurrentGlobalConnection(objectExplorerService, connectionManagementService, workbenchEditorService);
 		}
-		let configurationService = accessor.get<IConfigurationService>(IConfigurationService);
-		let previewFeaturesEnabled: boolean = configurationService.getValue('workbench')['enablePreviewFeatures'];
-		if (!previewFeaturesEnabled) {
-			return accessor.get<INotificationService>(INotificationService).info(nls.localize('backup.isPreviewFeature', 'You must enable preview features in order to use backup'));
+		if (profile) {
+			let serverInfo = connectionManagementService.getServerInfo(profile.id);
+			if (serverInfo && serverInfo.isCloud) {
+				return accessor.get<INotificationService>(INotificationService).info(nls.localize('backup.commandNotSupported', 'Backup command is not supported for Azure Sql databases.'));
```

Comment:
should 'Sql' be spelled `SQL`?

Reply:
Good point.. also should it be SQL Azure or Azure SQL :)? The former right?

Reply:
ok no, its **Azure SQL** at most places... so will keep that order..
![image](https://user-images.githubusercontent.com/46980425/58665597-16e97b80-82e6-11e9-9c50-5e5b641a385d.png)